Examples/Swift/MapSnapshotterExample.swift

```diff
@@ -0,0 +1,51 @@
+import Mapbox
+
+@objc(MapSnapshotterExample_Swift)
+
+class MapSnapshotterExample: UIViewController, MGLMapViewDelegate {
+
+    var mapView : MGLMapView!
+    var button : UIButton!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        mapView = MGLMapView(frame: view.bounds, styleURL: MGLStyle.satelliteStreetsStyleURL())
+        mapView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        
+        // Center map on the Giza Pyramid Complex in Egypt.
+        let center = CLLocationCoordinate2D(latitude: 29.9773, longitude: 31.1325)
+        mapView.setCenter(center, zoomLevel: 14, animated: false)
+        view.addSubview(mapView)
+
+        // Create a button to take a map snapshot.
+        button = UIButton(frame: CGRect(x: view.bounds.width / 2 - 15, y: view.bounds.height - 80, width: 30, height: 30))
+        button.layer.cornerRadius = 15
+        button.setImage(UIImage(named: "camera"), for: .normal)
+        button.addTarget(self, action: #selector(createSnapshot), for: .allTouchEvents)
+        view.addSubview(button)
+    }
+    
+    @objc func createSnapshot() {
+        
+        // Create a UIImageView that will store the map snapshot.
+        let imageView = UIImageView(frame: view.bounds)
```

Comment:
I commend you for keeping this example focused and relatively simple, but replacing the map with an almost identical image view isn’t very intuitive (and is somewhat difficult to screen-cap for the site, I’d imagine).

Could we juice this up a bit — maybe some sort of animation or effect during the map → snapshot transition that makes it clear that something is happening. Another option might be not having the image view cover the entire viewport.

Reply:
Something like [`UIImageWriteToSavedPhotosAlbum()`](https://developer.apple.com/documentation/uikit/1619125-uiimagewritetosavedphotosalbum) is another option, though it may require a few other changes to the project.

Reply:
Makes sense 👍  I originally was playing with having a map view in the top half of the view, then putting the map snapshot in a view in the lower half. I can go back to doing that.